src/ol/interaction/selectinteraction.js

```diff
@@ -201,6 +210,20 @@ goog.inherits(ol.interaction.Select, ol.interaction.Interaction);
 
 
 /**
+ * @param {ol.Feature} feature Feature.
+ * @param {ol.layer.Layer} layer Layer.
+ * @private
+ */
+ol.interaction.Select.prototype.addFeatureLayerAssociation_ =
+    function(feature, layer) {
+  var key = goog.getUid(feature);
+  var obj = {};
+  obj[key] = layer;
+  this.featureLayerAssociation_.push(obj);
```

Comment:
This could be handled much more efficiently if `this.featureLayerAssociation_` was an object:
```js
this.featureAssociation_[key] = layer;
```